packages/syft/src/syft/client/api.py

```diff
@@ -384,6 +384,9 @@ def _repr_html_(self) -> Any:
         results = self.get_all()
         return results._repr_html_()
 
+    def __len__(self) -> int:
+        return len(self.get_all())
+
```

Comment:
I think we should not add this here. This is an expensive call to get `__len__` and it's not obvious to users. They could just do `len(client...get_all())` instead.

Reply:
Not sure if it's expensive since we use it `_repr_html` and `__getitem__` so it gets called all the time
![image](https://github.com/OpenMined/PySyft/assets/88959106/244d0c7a-3762-488c-9c16-85cd1e377fb1)

Reply:
And I add this as it makes a lot of sense to do something like `len(client.images)` or `len(client.worker_pools)` to see how many are there

Reply:
> I think this depends on the `API.get_all()`, but we already use it `_repr_html` and `__getitem__` so it gets called all the time

It's true. These are also expensive, and I plan to discuss with the product team about this. Doesn't mean we should just keep adding to it in the mean time though. Plus this is beyond the scope of this PR. You're not just adding to `client.images` or `client.worker_pools` but all other endpoints (e.g. `client.datasets`) as well.

Reply:
Thanks @kiendang. This makes sense. I removed it